component/src/test/java/org/wso2/carbon/identity/authenticator/reddit/RedditTest.java

```diff
@@ -0,0 +1,217 @@
+    private AuthenticatedUser authenticatedUser;
+    @Spy
+    private AuthenticationContext context = new AuthenticationContext();
+    @Mock
+    private HttpServletResponse httpServletResponse;
+
+    @Mock
+    private OAuthClient mockOAuthClient;
+
+    @Mock
+    private OAuthClientRequest mockOAuthClientRequest;
+    @Mock
+    private OAuthJSONAccessTokenResponse oAuthJSONAccessTokenResponse;
+
+    @DataProvider(name = "authenticatorProperties")
+    public Object[][] getAuthenticatorPropertiesData() {
+
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_ID, "test-client-id");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_SECRET, "test-client-secret");
+        authenticatorProperties.put("callbackUrl", "http://localhost:9443/commonauth");
+        authenticatorProperties.put("scope", "");
+        return new Object[][]{{authenticatorProperties}};
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    RedditAuthenticator redditAuthenticator;
+
+    @BeforeMethod
+    public void setUp() {
+
+        redditAuthenticator = new RedditAuthenticator();
+        initMocks(this);
+    }
+
+    @Test(description = "Test case for getTokenEndpoint method", dataProvider = "authenticatorProperties")
+    public void testGetTokenEndpoint(
+            Map<String, String> authenticatorProperties) {
+
+        String tokenEndpoint = redditAuthenticator.getTokenEndpoint(authenticatorProperties);
+        Assert.assertEquals(RedditAuthenticatorConstants.REDDIT_TOKEN_ENDPOINT, tokenEndpoint);
+    }
+
+    @Test(description = "Test case for getUserInfoEndpoint method", dataProvider = "authenticatorProperties")
+    public void testGetUserInfoEndpoint(
+            Map<String, String> authenticatorProperties) {
+
+        String tokenEndpoint = redditAuthenticator.getUserInfoEndpoint(oAuthClientResponse, authenticatorProperties);
+        Assert.assertEquals(RedditAuthenticatorConstants.REDDIT_USERINFO_ENDPOINT, tokenEndpoint);
+    }
+
+    @Test(description = "Test case for requiredIDToken method", dataProvider = "authenticatorProperties")
+    public void testRequiredIDToken(
+            Map<String, String> authenticatorProperties) {
+
+        Assert.assertFalse(redditAuthenticator.requiredIDToken(authenticatorProperties));
+    }
+
+    @Test(description = "Test case for getFriendlyName method")
+    public void testGetFriendlyName() {
+
+        Assert.assertEquals(RedditAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME,
+                redditAuthenticator.getFriendlyName());
+    }
+
+    @Test(description = "Test case for getAuthorizationServerEndpoint method", dataProvider = "authenticatorProperties")
+    public void testGetAuthorizationServerEndpoint(
+            Map<String, String> authenticatorProperties) {
```

Comment:
Please do formatting and check this throughout the PR

Reply:
fixed with 2fe50e9